server/server.go

```diff
@@ -219,15 +220,13 @@ func handleHealth(w http.ResponseWriter, r *http.Request) {
 }
 
 func handleWebsocket(w http.ResponseWriter, r *http.Request) {
-	userId := getUserIdFromCookie(r)
-	responseHeader := setUserIdToCookie(w, userId)
-
-	if len(userId) == 0 {
-		userId = NewUserId()
-		responseHeader = setUserIdToCookie(w, userId)
+	userId := NewUserId()
+	if cookie, err := r.Cookie(cookiename); err == nil {
```

Comment:
Ooohh I think I didn't see `r.Cookie` in the docs. Is this somehow preferable to `r.Header.Get("Cookie")`?

Reply:
Yep! `http.Request.Cookie` parses the header and extracts the appropriate cookie by name. The [syntax of the Cookie header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cookie) is a little different from a URL query string, so the previous version using `net/url.ParseQuery` won't work if the browser ever sends more than one cookie.